chainservice/chainservice.go

```diff
@@ -96,11 +98,13 @@ func New(
 ) (*ChainService, error) {
 	// create indexers
 	var (
-		indexers         []blockdao.BlockIndexer
-		indexer          blockindex.Indexer
-		candidateIndexer *poll.CandidateIndexer
-		err              error
-		ops              optionParams
+		indexers                 []blockdao.BlockIndexer
```

Comment:
1. should implement as blockdao.BlockIndexer and add into here
2. can combine into 1 "stakingIndexer"